### PR TITLE
Command_spectator

### LINKED ISSUE #856: Implement /spectator and /gmsp
@OxLemonxO can probably do that for you, but I want to implement this because _FOR ADMINS_, it can be useful and try to trick the player on that we arent really there and him get caught and & banned for griefing. It's also useful to just check on players instead of them saying "GO AWAY" on you.

### PR DESCRIPTION
**Pull request by LegendIsAwesomes**
Resolves #856

### CODE REVIEW

**Comment by LegendIsAwesomes** [comment]
Tested & works @JeromSar

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import me.StevenLawson.TotalFreedomMod.TFM_AdminList;
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.bukkit.GameMode;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.OP, source = SourceType.BOTH)
+@CommandParameters(description = "Change yours or another players gamemode to Spectator Mode.", usage = "/<command> [partialname] | [-a]", aliases = "spec")
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java]: `gmc` should be an alias.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package me.StevenLawson.TotalFreedomMod.Commands;
+
+import me.StevenLawson.TotalFreedomMod.TFM_AdminList;
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.bukkit.GameMode;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.OP, source = SourceType.BOTH)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java]: Please make this command admin only.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            {
+                if (!TFM_AdminList.isSuperAdmin(sender))
+                {
+                    sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+                    return true;
+                }
+
+                for (Player targetPlayer : server.getOnlinePlayers())
+                {
+                    targetPlayer.setGameMode(GameMode.CREATIVE);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java]: Wrong gamemode.

**Comment by LegendIsAwesomes** [comment]
Why should gmc be a aliases, this is spectator not creative.

**Comment by LifeAsStranger** [comment]
@LegendIsAwesomes, he meant gmsp or something similar.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java (hunk 3 above)
Comment (LifeAsStranger) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java]: Should be ```java targetPlayer.setGameMode(GameMode.SPECTATOR); ```

**Comment by CrafterSmith12** [comment]
I already have this.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        Player player;
+        if (args.length == 0)
+        {
+            player = sender_p;
+        }
+        else
+        {
+            if (args[0].equalsIgnoreCase("-a"))
+            {
+                if (!TFM_AdminList.isSuperAdmin(sender))
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java]: don't need this check

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                {
+                    targetPlayer.setGameMode(GameMode.SPECTATOR);
+                }
+
+                TFM_Util.adminAction(sender.getName(), "Changing everyone's gamemode to spectator mode.", false);
+                return true;
+            }
+            
+            if (!(senderIsConsole || TFM_AdminList.isSuperAdmin(sender)))
+            {
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java]: don't need this check

**Pull request closed without merging**